01_stratified_classification/scripts/run_lasso_compare_vogelstein.sbatch

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+#SBATCH -N 1
+#SBATCH -t 0-24:00
+#SBATCH --array=0-124%15
+#SBATCH --cpus-per-task=8
+#SBATCH --output=slurm_output/slurm-%A_%a.out
+#SBATCH --error=slurm_output/slurm-%A_%a.err
+
+# activate conda env
+eval "$(conda shell.bash hook)"
+conda activate ../mpmp-env2
+echo "Environment loaded"
+
+# read the list of cosmic genes from tsv file
+merged_filename="data/vogelstein_cancergenes.csv"
+
+read_genes_from_file() {
+    # create global gene array
+    declare -a -g genes
+
+    # read tab-separated file, genes should be the first column
+    while IFS=$'\t' read -r gene class; do
+        genes+=("$gene")
+    done < "$1"
+
+    # remove header
+    genes=("${genes[@]:1}")
+}
+read_genes_from_file $merged_filename
+
+gene=${genes[${SLURM_ARRAY_TASK_ID}]}
+
+LL_RESULTS_DIR=./01_stratified_classification/results/optimizer_compare_ll
+SGD_RESULTS_DIR=./01_stratified_classification/results/optimizer_compare_sgd
+
+# number of features to "preselect" to by mean absolute deviation
+# -1 == no preselection
+MAD_PRESELECT=16042
```

Comment:
What's the logic behind this number?

Reply:
That's the number of genes in the gene set, that we have gene expression values for. My scripts mostly default to 8000 (we were using this for my past paper and other work, partially for computational reasons and partially because that's what Greg's mutation classification paper used, so we wanted our models to be comparable to those).

For this project I think it's a bit more important to use all the genes, since we're interested in model sparsity the background set for that should probably be all the genes in the gene set rather than a subset of them. I'll make a note of this in this script.